internal/services/web/app_service_certificate_resource.go

```diff
@@ -241,11 +244,12 @@ func resourceAppServiceCertificateSchema() map[string]*pluginsdk.Schema {
 		},
 
 		"key_vault_id": {
-			Type:         pluginsdk.TypeString,
-			Optional:     true,
-			ForceNew:     true,
-			ValidateFunc: commonids.ValidateKeyVaultID,
-			RequiredWith: []string{"key_vault_secret_id"},
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			ForceNew:         true,
+			DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
how come we are suppressing the case difference here now?

if there is a valid reason to it should be documented why, and if its due to an API bug should be linked to a rest api specs issue

Reply:
updated to add the link to github issue, it actually exists for a long time I suppose. But the test cases ignored them previously..